src/System.Net.Security/src/System/Net/SecureProtocols/AuthenticatedStream.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace System.Net.Security
+{
+    // A public contact for a base abstract authenticated stream.
+    public abstract class AuthenticatedStream : Stream
+    {
+        private Stream _InnerStream;
+        private bool _LeaveStreamOpen;
+
+        protected AuthenticatedStream(Stream innerStream, bool leaveInnerStreamOpen)
+        {
+            if (innerStream == null || innerStream == Stream.Null)
+            {
+                throw new ArgumentNullException("innerStream");
+            }
+
+            if (!innerStream.CanRead || !innerStream.CanWrite)
+            {
+                throw new ArgumentException(SR.net_io_must_be_rw_stream, "innerStream");
+            }
+
+            _InnerStream = innerStream;
+            _LeaveStreamOpen = leaveInnerStreamOpen;
+        }
+
+        public bool LeaveInnerStreamOpen
+        {
+            get
+            {
+                return _LeaveStreamOpen;
+            }
+        }
+
+        protected Stream InnerStream
+        {
+            get
+            {
+                return _InnerStream;
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+#if DEBUG
+            using (GlobalLog.SetThreadKind(ThreadKinds.User))
+            {
+#endif
+                try
+                {
+                    if (disposing)
+                    {
+                        if (_LeaveStreamOpen)
+                        {
+                            _InnerStream.Flush();
+                        }
+                        else
+                        {
+                            _InnerStream.Dispose();
```

Comment:
We're not flushing in this case... I guess the assumption is that Dispose on the innerStream will flush itself if it needs to?